api/src/test/java/com/epam/pipeline/util/CustomAssertions.java

```diff
@@ -61,4 +61,20 @@ public static void assertThrows(final Runnable runnable) {
         }
         fail("Exception was expected but nothing was thrown.");
     }
+
+    public static void assertThrowsChecked(final Class<? extends Throwable> expectedExceptionClass,
+                                           final CheckedRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Exception e) {
+            if (expectedExceptionClass.isInstance(e)) {
+                return;
+            } else {
+                throw new AssertionError(
+                        String.format("Expected exception %s was not thrown, but another exception was: %s.",
+                                expectedExceptionClass.getSimpleName(), e.getClass().getSimpleName()), e);
+            }
+        }
+        fail(String.format("Expected exception %s was not thrown.", expectedExceptionClass.getSimpleName()));
+    }
```

Comment:
I imagine if we can reuse `assertThrows` method here using `@SneakyThrows` annotation. The call site will look like this:
```
assertThrows(expectedExceptionClass, sneakyThrowing(runnable));
```

Reply:
I support that something like this should work:
```java
    @SneakyThrows
    public static void sneakyThrowing(final CheckedRunnable runnable) {
        return runnable.run()
    }

    public static void assertThrowsChecked(final Class<? extends Throwable> expectedExceptionClass,
                                           final CheckedRunnable runnable) {
        assertThrows(expectedExceptionClass, () -> sneakyThrowing(runnable));
    }
```